lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py

```diff
@@ -1253,10 +1268,16 @@ def create_default_nic(self):
                              "machine.".format(virtual_network_name)
 
             subnet_id = None
-            try:
-                subnets = self.network_client.subnets.list(self.resource_group, virtual_network_name)
-            except CloudError:
-                self.fail(no_subnets_msg)
+            if self.virtual_network_resource_group:
+                try:
+                    subnets = self.network_client.subnets.list(self.virtual_network_resource_group, virtual_network_name)
```

Comment:
No need to duplicate these lines of code- just conditionally assign a local `virtual_network_resource_group` as `self.resource_group` or `self.virtual_network_resource_group` as appropriate and use it below the conditional.